client/components/time-since/index.jsx

```diff
@@ -1,40 +1,32 @@
-import moment from 'moment';
-import { PureComponent } from 'react';
-import humanDate from 'calypso/lib/human-date';
-import { Interval, EVERY_TEN_SECONDS } from 'calypso/lib/interval';
-import smartSetState from 'calypso/lib/react-smart-set-state';
+import { useState, useEffect, useMemo } from 'react';
+import { useLocalizedMoment } from 'calypso/components/localized-moment';
+import getHumanDate from 'calypso/lib/human-date';
+import { useInterval, EVERY_TEN_SECONDS } from 'calypso/lib/interval';
 
-export default class TimeSince extends PureComponent {
-	smartSetState = smartSetState;
+function TimeSince( { className, date, dateFormat } ) {
+	const [ humanDate, setHumanDate ] = useState( '' );
+	const moment = useLocalizedMoment();
+	const fullDate = useMemo( () => moment( date ).format( 'llll' ), [ moment, date ] );
 
-	UNSAFE_componentWillMount() {
-		this.update();
-	}
+	function update() {
+		const newHumanDate = getHumanDate( date, dateFormat );
 
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		this.update( nextProps.date );
+		if ( newHumanDate !== humanDate ) {
+			setHumanDate( newHumanDate );
+		}
 	}
 
-	update = ( date ) => {
-		const { dateFormat } = this.props;
-		date = date || this.props.date;
+	useInterval( update, EVERY_TEN_SECONDS );
 
-		this.smartSetState( {
-			humanDate: humanDate( date, dateFormat ),
-			fullDate: moment( date ).format( 'llll' ),
-		} );
-	};
+	useEffect( () => {
+		update();
+	}, [] );
```

Comment:
Instead of an effect, the initial update can be done when initializing state:
```js
useState( () => getHumanDate( getHumanDate( date, dateFormat ) ) );
```
Then `humanDate` will have the desired value (not a `''`) right in the very first render. That's more efficient (one render instead of two) and makes it also work in SSR.

One shortcoming, both in the `useEffect` and `useState` versions, is that `humanDate` will not correctly react to changes in the `date` and `dateFormat` props. The rendered `humanDate` will be out of sync until the next interval tick that fixes the value.

That's something that the `useHumanDate` hook should address.